### PR TITLE
DateTime Implementation

### PR DESCRIPTION
**Pull request by naqvisat**
DateTime Implementation.

### CODE REVIEW

File: shared/src/main/scala/pl/metastack/metatime/DateTime.scala [hunk 1]
```diff
@@ -1,18 +1,87 @@
...
 }
 
 object DateTime {
-  def now(): DateTime = ???
-  def apply(year: Int, month: Int, day: Int): DateTime = ???
+
+  def calculateDateTime(milliDate: Long, milliTime: Long): DateTime = {
+    val date = Date(milliDate)
+    val time = Time(milliTime)
+    DateTime(date.year, date.month, date.day, time.h, time.m, time.s, time.ms).asInstanceOf[DateTime]
```
Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/DateTime.scala]: The cast is not needed.

File: shared/src/main/scala/pl/metastack/metatime/DateTime.scala [hunk 2]
```diff
@@ -1,18 +1,87 @@
...
+    override val h: Int = hr
+    override val m: Int = min
+    override val s: Int = sec
+    override val ms: Int = milli
+  }
+
+  def apply(comp : Component) : DateTime = {
+    comp match {
+      case dT: DateTime =>
+        DateTime(dT.year, dT.month, dT.day, dT.h, dT.m, dT.s, dT.ms).asInstanceOf[DateTime]
```
Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/DateTime.scala]: The casts can be removed.

File: shared/src/main/scala/pl/metastack/metatime/DateTime.scala (hunk 1 above)
Comment (naqvisat) [inline shared/src/main/scala/pl/metastack/metatime/DateTime.scala]: Absolute I agree but without it IntellijIDEA gives this error: "Expression of type Date doesn't conform to expected type DateTime". Any idea how can I get rid of this error?

File: shared/src/main/scala/pl/metastack/metatime/DateTime.scala (hunk 2 above)
Comment (naqvisat) [inline shared/src/main/scala/pl/metastack/metatime/DateTime.scala]: Absolute I agree but without it IntellijIDEA gives this error: "Expression of type Date doesn't conform to expected type DateTime". Any idea how can I get ride of this error?

File: shared/src/main/scala/pl/metastack/metatime/DateTime.scala (hunk 1 above)
Comment (tindzk) [inline shared/src/main/scala/pl/metastack/metatime/DateTime.scala]: Change the `apply()` method to return the right type (`DateTime` instead of `Date`).

Comment (naqvisat) [inline shared/src/main/scala/pl/metastack/metatime/DateTime.scala]: Ouch, Thanks ;D

**Pull request merged**